### ISSUE TITLE
eddy_cuda not working

### ISSUE THREAD

**Issue by soichih** [brain-life/app-mrtrix3-preproc#3]
@bcmcpher I've *fixed* the issue with eddy_cuda so that it will run with my test data inside the container used by the App. However, when I try to run it via mrtrix3_preproc.sh, it's still failing over to eddy_opemmp.

I've remove -quiet with set +x to show more output, but I am still not seeing much info.. I wonder if some of the options set for dwipreproc is not compatible with eddy_cuda?

```
+ dwipreproc -eddy_options ' --repol --data_is_shelled --slm=linear' -rpe_none -pe_dir PA dwi_denoise_degibbs.mif dwi_denoise_degibbs_eddy.mif -tempdir ./tmp -nthreads 8
dwipreproc: 
dwipreproc: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
dwipreproc: 
dwipreproc: Generated temporary directory: /media/data/brlife/workdirs/5ba702271600c3002abd97d0/5ba7c9e61600c3002abd9ac0/tmp/dwipreproc-tmp-GRJAYN/
Command:  mrconvert /media/data/brlife/workdirs/5ba702271600c3002abd97d0/5ba7c9e61600c3002abd9ac0/dwi_denoise_degibbs.mif /media/data/brlife/workdirs/5ba702271600c3002abd97d0/5ba7c9e61600c3002abd9ac0/tmp/dwipreproc-tmp-GRJAYN/dwi.mif
dwipreproc: Changing to temporary directory (/media/data/brlife/workdirs/5ba702271600c3002abd97d0/5ba7c9e61600c3002abd9ac0/tmp/dwipreproc-tmp-GRJAYN/)
dwipreproc: Total readout time not provided at command-line; assuming sane default of 0.1
Command:  mrinfo dwi.mif -export_grad_mrtrix grad.b
Command:  dwi2mask dwi.mif - | maskfilter - dilate - | mrconvert - eddy_mask.nii -datatype float32 -strides -1,+2,+3
Command:  mrconvert dwi.mif -import_pe_table dwi_manual_pe_scheme.txt eddy_in.nii -strides -1,+2,+3,+4 -export_grad_fsl bvecs bvals -export_pe_eddy eddy_config.txt eddy_indices.txt
Command:  eddy_cuda --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --repol --data_is_shelled --slm=linear --out=dwi_post_eddy
dwipreproc: [WARNING] Command failed: eddy_cuda --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --repol --data_is_shelled --slm=linear --out=dwi_post_eddy
dwipreproc: [WARNING] CUDA version of eddy appears to have failed; trying OpenMP version
Command:  eddy_openmp --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --repol --data_is_shelled --slm=linear --out=dwi_post_eddy

```

Can we try troubleshooting this together? I also think we shouldn't be using eddy_cuda to begin with. Let's discuss.

**Comment by francopestilli** [comment]
eddy_cuda might not be the best option as it is unlikely that we will have gpu on most systems.
Perhaps we can make it optional for this App?

**Comment by soichih** [comment]
I got it working. I had to set absolute path for nvidia-340/cuda library as dwipreproc changes currently directory to a temp dir internally.

@francopestilli This App is registered to run on gpu1-pestillilab.psych.indiana.edu as well as other HCP clusters. If this App runs on gpu1, it will run much faster by using the GPU on that machine. You are correct that we currently don't have access to many GPU enabled machine (we should submit a requet to have access to XSEDE / Bridges or Comets GPU clusters for the next year's jetstream allocation?) but I'd like to make sure that we have a few App that runs on our GPU machines so that we can demonstrate the capability of Brainlife to run various Apps on various resources that each Apps are most appropriate to run on.
